tmpl/wiki/package-quality.md

```diff
@@ -0,0 +1,138 @@
+gaps in the ecosystem, but things are getting better!
+
+The next problem is knowing the level of confidence to attribute to each of
+these libraries. Would be great to be able to answer questions such as:
+Does the library has a good test coverage? Was it formally verified?
+Was it used in production? In order to do this, I would like to introduce
+the "MirageOS quality levels". Ideally this will live in a tool (e.g.
+`mirage lint`) but to start we list the criteria that package aiming
+for increased quality should follow.
+
+### ![Level 1:](https://img.shields.io/badge/level-1-blue.svg) Follow Packaging Guidelines
+
+See the [packaging guidelines](https://mirage.io/wiki/packaging) for more
+details. In summary:
+
+- The package should work well with [odig](http://erratique.ch/software/odig)
+  e.g. `CHANGES.md`, `LICENSE.md` and `README.md` should exist.
+
+- The package should work well with [topkg](http://erratique.ch/software/topkg),
+  e.g. `pkg/pkg.ml` should exist.
+
+- The use of [jbuilder](http://jbuilder.readthedocs.io/en/latest/) is
+  strongly encouraged but not yet mandatory as 1.0.0 has not been
+  officially released. However, we plan to enforce its use as soon as
+  it is the case -- assuming that it has all the features needed for
+  building MirageOS applications.
+
+- Use proper names: same name for opam and ocamlfind package, and also ideally
+  for the top-level module.
+
+- All direct dependencies are specified in the opam file (e.g. sometimes some
+  direct dependencies are included by transitivity -- this is not very stable
+  and could break the package in the future if one of a transitive dependency
+  is updated) (NOTE: should probably be added in the packaging guideline)
+
+### ![Level 2](https://img.shields.io/badge/level-2-blue.svg) Define Package Scope
+
+- Have well-identified maintainers. GitHub recently
+  [introduced code owners](https://github.com/blog/2392-introducing-code-owners),
+  which may be worth to use for some (of our core) libraries.
+
+- Have a design document specifying the scope of the library (and esp. what is
+  out of scope)
+
+- Use the correct set of tags in opam metadata
+
+- Have documentation
+
+### ![Level 3:](https://img.shields.io/badge/level-3-blue.svg) Use Good Coding Style
+
+- Have proper indentation (using ocp-indent + checked-in ocp-indent file).
+
+- Use at most 80 columns (`ocp-indent` unfortunately doesn't check this).
+
+- Follow [OCaml programming guidelines](https://ocaml.org/learn/tutorials/guidelines.html).
+
+### ![Level 4:](https://img.shields.io/badge/level-4-blue.svg) Keep your Style Functional
+
+- Avoid global mutable state.
+
+- Use tail recursion.
+
+- Use `-safe-string` (looks like this will be the default in 4.06).
+
+- Avoid polymorphic equality and comparison -- when possible define your own
+  specialized functions.
+
+### ![Level 5:](https://img.shields.io/badge/level-5-blue.svg) Test
+
+- Have unit tests (using alcotest) (ideally with coverage report).
```

Comment:
Automated tests for whether the library actually works with `mirage` (current `HEAD` of the primary branch probably, rather than the last release) would be good to see somewhere in here, although I'm not sure where the best place is.